experimental/dds/tree2/src/feature-libraries/typed-schema/typedTreeSchema.ts

```diff
@@ -362,7 +363,11 @@ export class TreeFieldSchema<
 	/**
 	 * This is computed lazily since types can be recursive, which makes evaluating this have to happen after all the schema are defined.
 	 */
-	private readonly lazyTypes: Lazy<{ names: TreeTypeSet; schema: AllowedTypeSet }>;
+	private readonly lazyTypes: Lazy<{
+		names: TreeTypeSet;
+		schema: AllowedTypeSet;
+		monomorphicChildType?: TreeNodeSchema;
```

Comment:
Could be worth adding a comment on this field indicating why it exists.

Reply:
monomorphicChildType, specifically

Reply:
I added details to the comment on the public API below to cover this.